### PR TITLE
Change minion to node

### PR DESCRIPTION
**Pull request by duglin**
Continuation of #1111

I tried to keep this PR down to just a simple search-n-replace to keep
things simple.  I may have gone too far in some spots but its easy to
roll those back if needed - just let me know.

I avoided renaming `contrib/mesos/pkg/minion` because there's already
a `contrib/mesos/pkg/node` dir and fixing that will require a bit of work
due to a circular import chain that pops up. So I'm saving that for a
follow-on PR.

Signed-off-by: Doug Davis dug@us.ibm.com

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/25260)

### CODE REVIEW

**Comment by lavalamp** [comment]
The salt changes make me nervous. Some of the other changes, too, but I didn't look carefully. Changes in documentation and comments should be OK.

**Comment by duglin** [comment]
@lavalamp nervous enough that I should roll them back or do you want to wait until a deeper review and the testcases are run?

**Comment by lavalamp** [comment]
ok to test

**Comment by lavalamp** [comment]
Well, let's see if they pass the tests first.

File: docs/api.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ Some important differences between v1beta1/2 and v1beta3:
 * The resource `id` is now called `name`.
 * `name`, `labels`, `annotations`, and other metadata are now nested in a map called `metadata`
 * `desiredState` is now called `spec`, and `currentState` is now called `status`
-* `/minions` has been moved to `/nodes`, and the resource has kind `Node`
+* `/nodes` has been moved to `/nodes`, and the resource has kind `Node`
```
Comment (lavalamp) [inline docs/api.md]: Maybe find&replace is a little overeager :)

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf [hunk 2]
```diff
@@ -123,7 +123,7 @@
   type tail
   format /^(?<time>[^ ]* [^ ,]*)[^\[]*\[[^\]]*\]\[(?<severity>[^ \]]*) *\] (?<message>.*)$/
   time_format %Y-%m-%d %H:%M:%S
-  path /var/log/salt/minion
+  path /var/log/salt/node
```
Comment (lavalamp) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf]: not certain what this does and whether it should be changed.

File: cluster/aws/config-default.sh [hunk 3]
```diff
@@ -76,13 +76,13 @@ MASTER_DISK_SIZE=${MASTER_DISK_SIZE:-20}
...
 MASTER_ROOT_DISK_SIZE=${MASTER_ROOT_DISK_SIZE:-8}
-# The minions root EBS volume size (used to house Docker images)
+# The nodes root EBS volume size (used to house Docker images)
 NODE_ROOT_DISK_TYPE="${NODE_ROOT_DISK_TYPE:-gp2}"
 NODE_ROOT_DISK_SIZE=${NODE_ROOT_DISK_SIZE:-32}
 
 MASTER_NAME="${INSTANCE_PREFIX}-master"
 MASTER_TAG="${INSTANCE_PREFIX}-master"
-NODE_TAG="${INSTANCE_PREFIX}-minion"
+NODE_TAG="${INSTANCE_PREFIX}-node"
```
Comment (lavalamp) [inline cluster/aws/config-default.sh]: This sort of thing could break existing clusters if people ran *-up.sh one way and then want to run a different script the other way.

File: cluster/aws/templates/configure-vm-aws.sh [hunk 4]
```diff
@@ -79,7 +79,7 @@ fix-apt-sources() {
 }
 
 salt-master-role() {
-  cat <<EOF >/etc/salt/minion.d/grains.conf
+  cat <<EOF >/etc/salt/node.d/grains.conf
```
Comment (lavalamp) [inline cluster/aws/templates/configure-vm-aws.sh]: I'm not sure about this change either.

**Comment by lavalamp** [comment]
I left some comments on the first few concerning things that jumped out at me, but did not do a complete pass.

**Comment by duglin** [comment]
@lavalamp thanks for the comments - kind of swamped right now but will look at very soon.
In the meantime, I've rebased.

**Comment by jfrazelle** [comment]
rebase/update?

**Comment by duglin** [comment]
rebased

**Comment by duglin** [comment]
@k8s-bot gce e2e test this

**Comment by duglin** [comment]
Scaled this PR back quite a bit to make it easier to get thru the testing. All are green now!
Will do more PRs for other "minion"s later.

**Comment by duglin** [comment]
rebased - and ready for review

**Comment by duglin** [comment]
@k8s-bot gke e2e test this

**Comment by duglin** [comment]
@k8s-bot gke e2e test this

**Comment by duglin** [comment]
quick its green!! :-)
ping @jfrazelle

**Comment by jfrazelle** [comment]
@duglin got your back :) LGTM

**Comment by duglin** [comment]
@jfrazelle woo hoo - thanks!

**Pull request merged**